src/libraries/System.Linq/src/System/Linq/DefaultIfEmpty.cs

```diff
@@ -8,7 +8,7 @@ namespace System.Linq
 {
     public static partial class Enumerable
     {
-        public static IEnumerable<TSource> DefaultIfEmpty<TSource>(this IEnumerable<TSource> source) =>
+        public static IEnumerable<TSource?> DefaultIfEmpty<TSource>(this IEnumerable<TSource> source) =>
             DefaultIfEmpty(source, default!);
 
         public static IEnumerable<TSource> DefaultIfEmpty<TSource>(this IEnumerable<TSource> source, TSource defaultValue)
```

Comment:
>TSource [](start = 101, length = 7)

`TSource? defaultValue`?

Reply:
If the developer wants to pass in a null default value, they can use a nullable type; we don't need to force nullable on all uses.